File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/CommentBasedSnippetVariableResolver.java

package io.github.abc15018045126.sora.widget.snippet.variable;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class CommentBasedSnippetVariableResolver implements ISnippetVariableResolver {

    private String[] commentTokens;

    public CommentBasedSnippetVariableResolver() {
        this(null);
    }

    /**
     * For example, Java language: {@code new String[]{ "//", "/*", "*\/" }
     */
    public CommentBasedSnippetVariableResolver(@Nullable String[] commentTokenStrings) {
        setCommentTokens(commentTokenStrings);
    }

    public void setCommentTokens(@Nullable String[] commentTokens) {
        this.commentTokens = commentTokens;
    }

    public String[] getCommentTokens() {
        return commentTokens;
    }

    @NonNull
    @Override
    public String[] getResolvableNames() {
        return new String[]{
                "LINE_COMMENT", "BLOCK_COMMENT_START", "BLOCK_COMMENT_END"
        };
    }

    @NonNull
    @Override
    public String resolve(@NonNull String name) {
        if (commentTokens == null || commentTokens.length != 3) {
            throw new IllegalStateException("language comment style is not configured properly");
        }
        switch (name) {
            case "LINE_COMMENT":
                return commentTokens[0];
            case "BLOCK_COMMENT_START":
                return commentTokens[1];
            case "BLOCK_COMMENT_END":
                return commentTokens[2];
        }
        throw new IllegalArgumentException("Unsupported variable name:" + name);
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/CommentBasedSnippetVariableResolver.kt
package io.github.abc15018045126.sora.widget.snippet.variable

class CommentBasedSnippetVariableResolver(var commentTokens: Array<String>? = null) : ISnippetVariableResolver {

    override fun getResolvableNames(): Array<String> {
        return arrayOf(
            "LINE_COMMENT", "BLOCK_COMMENT_START", "BLOCK_COMMENT_END"
        )
    }

    override fun resolve(name: String): String {
        val tokens = commentTokens
        if (tokens == null || tokens.size != 3) {
            throw IllegalStateException("language comment style is not configured properly")
        }
        return when (name) {
            "LINE_COMMENT" -> tokens[0]
            "BLOCK_COMMENT_START" -> tokens[1]
            "BLOCK_COMMENT_END" -> tokens[2]
            else -> throw IllegalArgumentException("Unsupported variable name:$name")
        }
    }
}
